portal/src/graphql/portal/ManageLanguageWidget.tsx

```diff
@@ -296,25 +329,53 @@ const ManageLanguageWidget: React.VFC<ManageLanguageWidgetProps> =
       setIsDialogPresented(false);
     }, []);
 
-    const templateLocaleOptions = useMemo(() => {
+    const templateLocaleOptions: IDropdownOption[] = useMemo(() => {
       const options = [];
-      for (const locale of supportedLanguagesForDropdown) {
+
+      const combinedLocales = new Set([
+        ...existingLanguages,
+        ...supportedLanguages,
+      ]);
+
+      for (const locale of combinedLocales) {
+        const isNew = !existingLanguages.includes(locale);
+        const isRemoved = !supportedLanguages.includes(locale);
+
+        let localeDisplay = displayTemplateLocale(locale);
+        if (isRemoved) {
+          localeDisplay = renderToString(
+            "ManageLanguageWidget.option-removed",
+            {
+              LANG: localeDisplay,
+            }
+          );
+        }
+
         options.push({
           key: locale,
-          text: displayTemplateLocale(locale),
+          text: localeDisplay,
           data: {
             isFallbackLanguage: fallbackLanguage === locale,
           },
+          disabled: isRemoved || isNew,
         });
       }
 
       return options;
     }, [
-      supportedLanguagesForDropdown,
+      existingLanguages,
+      supportedLanguages,
       displayTemplateLocale,
       fallbackLanguage,
+      renderToString,
     ]);
 
+    const hasNewLanguage = useMemo(() => {
```

Comment:
unused variable.

Reply:
It's used in prev commit but didn't get included, rearranged the commits